common/util/cmd_positional_arguments.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_COMMON_UTIL_CMD_POSITIONAL_ARGUMENTS_H_
+#define VERIBLE_COMMON_UTIL_CMD_POSITIONAL_ARGUMENTS_H_
+
+#include <string>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "absl/strings/string_view.h"
+
+namespace verible {
+
+class CmdPositionalArguments {
```

Comment:
We already have some functionality that extracts these information from a file, check out this:

https://github.com/chipsalliance/verible/blob/6b2724257d54225491ee48214fa08f4ddfb4d29c/verilog/analysis/verilog_project.h#L322-L340

The struct currently only contains files and incdirs, but we can easily add macro defines.

Maybe we can just move your functionality there and have another toplevel function (not even a class like this); we can call it

```c++
// Parse positional parameters from command line and extract files, +incdir+ and +define+
FileList ParseSourceFileListFromCommandline(std::vector<char *> cmdline);
```

That then will help us to merge the functionality.